crates/matrix-sdk-crypto/src/olm/utility.rs

```diff
@@ -50,56 +53,148 @@ impl SignJson for Ed25519SecretKey {
     }
 }
 
+/// A trait for objects that are able to verify signatures.
 pub trait VerifyJson {
-    /// Verify a signed JSON object.
+    /// Verify a signature over the SignedJsonObject using this public Ed25519
+    /// key.
     ///
-    /// The object must have a signatures key associated  with an object of the
-    /// form `user_id: {key_id: signature}`.
+    /// # Arguments
+    ///
+    /// * `user_id` - The user that claims to have signed this object.
+    ///
+    /// * `key_id` - The ID of the key that was used to sign this object.
+    /// **Note**: The key ID must match the ID of the public key that is
+    /// verifying the signature. This is only used to find the correct
+    /// signature.
+    ///
+    /// * `signed_object` - The signed object that we should check for a valid
+    /// signature.
     ///
     /// Returns Ok if the signature was successfully verified, otherwise an
     /// SignatureError.
+    fn verify_json(
+        &self,
+        user_id: &UserId,
+        key_id: &DeviceKeyId,
```

Comment:
Couldn't we calculate the key ID ourselves instead of passing it?

Reply:
No, not for devices, you'll need to pass the `DeviceId`. I think `Option<DeviceId>` would be worse here.